app/services/metrics.py

```diff
@@ -19,30 +17,60 @@ def get_areas_by_defined_area(area_type, area_id):
         # TODO: Implement service
         return ""
 
-    def get_areas_by_polygon(polygon):
-        # TODO: read raster from STAC
-        raster_cloud_path = "https://staccatalog.blob.core.windows.net/cog-test/Colombia_pp-2015_12_31-pp_2011_2015.tif"
+    def get_areas_by_polygon(
+        polygon: PolygonGeometry, metric_id: str
+    ) -> list[dict[str, float]]:
+        collection_url = find_collection_url(settings.stac_url, metric_id)
+        if not collection_url:
+            raise ValueError(f"No collection found for id: {metric_id}")
+
+        items_url = get_collection_items_url(collection_url)
+
+        if not items_url:
+            raise ValueError(
+                f"No items URL found for collection id: {metric_id}"
+            )
```

Comment:
these lines are repeated in both functions, and will be repeated in the services for defined areas too, the logic to get the items for a given collection should be handled in the `collection` file.